specification/monitor/resource-manager/Microsoft.Insights/stable/2023-10-01/metricDefinitions_API.json

```diff
@@ -0,0 +1,384 @@
+          }
+        }
+      }
+    },
+    "/{resourceUri}/providers/Microsoft.Insights/metricDefinitions": {
+      "get": {
+        "tags": [
+          "MetricDefinitions"
+        ],
+        "operationId": "MetricDefinitions_List",
+        "description": "Lists the metric definitions for the resource.",
+        "parameters": [
+          {
+            "$ref": "./commonMonitoringTypes.json#/parameters/ResourceUriParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "./commonMonitoringTypes.json#/parameters/MetricNamespaceParameter"
+          }
+        ],
+        "responses": {
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v5/types.json#/definitions/ErrorDetail"
+            }
+          },
+          "200": {
+            "description": "Successful request to get the list of metric definitions",
+            "schema": {
+              "$ref": "#/definitions/MetricDefinitionCollection"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": null
+        },
+        "x-ms-examples": {
+          "Get Metric Definitions without filter": {
+            "$ref": "./examples/GetMetricDefinitions.json"
+          },
+          "Get Application Insights Metric Definitions without filter": {
+            "$ref": "./examples/GetMetricDefinitionsApplicationInsights.json"
+          },
+          "Get StorageCache Metric Definitions with metric class": {
+            "$ref": "./examples/GetMetricDefinitionsMetricClass.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "MetricAvailability": {
+      "type": "object",
+      "properties": {
+        "timeGrain": {
+          "type": "string",
+          "format": "duration",
+          "description": "the time grain specifies the aggregation interval for the metric. Expressed as a duration 'PT1M', 'P1D', etc."
+        },
+        "retention": {
+          "type": "string",
+          "format": "duration",
+          "description": "the retention period for the metric at the specified timegrain.  Expressed as a duration 'PT1M', 'P1D', etc."
+        }
+      },
+      "description": "Metric availability specifies the time grain (aggregation interval or frequency) and the retention period for that time grain."
+    },
+    "AggregationType": {
+      "type": "string",
+      "description": "the aggregation type of the metric.",
+      "enum": [
+        "None",
+        "Average",
+        "Count",
+        "Minimum",
+        "Maximum",
+        "Total"
+      ],
+      "x-ms-enum": {
+        "name": "AggregationType",
+        "modelAsString": false
+      }
+    },
+    "MetricAggregationType": {
```

Comment:
Is it possible to reconcile AggregationType and MetricAggregationType? Seems redundant to have both.